crates/parser/src/grammar/expressions.rs

```diff
@@ -16,9 +16,9 @@ pub(super) enum Semicolon {
 
 const EXPR_FIRST: TokenSet = LHS_FIRST;
 
-pub(super) fn expr(p: &mut Parser<'_>) -> bool {
+pub(super) fn expr(p: &mut Parser<'_>) -> Option<CompletedMarker> {
     let r = Restrictions { forbid_structs: false, prefer_stmt: false };
-    expr_bp(p, None, r, 1).is_some()
+    expr_bp(p, None, r, 1).map(|(m, _)| m)
```

Comment:
I found that the `BlockLike` returned by `expr_bp` is only useful if `prefer_stmt` is `true`, otherwise it is always `NotBlock`, so I don't make this function reutrn `BlockLike`